app/scripts/controllers/service.js

```diff
@@ -27,11 +28,60 @@ angular.module('openshiftConsole')
       }
     ];
 
+    $scope.podFailureReasons = {
+      "Pending": "This pod will not receive traffic until all of its containers have been created."
+    };
+
+    var allPods = {};
     var watches = [];
 
+    // receives routes for the current service and maps service ports to each route name
+    var getPortsByRoute = function() {
+      if(!$scope.service) {
+        return;
+      }
+
+      $scope.portsByRoute = {};
+
+      _.each($scope.service.spec.ports, function(port) {
+        var reachedByRoute = false;
+        _.each($scope.routesForService, function(route) {
+          if(!route.spec.port || route.spec.port.targetPort === port.name ||
```

Comment:
@spadgett you are more familiar with the route / port stuff, can you review this section